internal/proxyservice/tcp/pg/backend.go

```diff
@@ -2,51 +2,46 @@ package pg
 
 import (
 	"fmt"
-	"log"
 	"net"
 	"reflect"
 
-	"github.com/cyberark/secretless-broker/internal/handlers/pg/protocol"
 	"github.com/cyberark/secretless-broker/internal/handlers/ssl"
-	"github.com/cyberark/secretless-broker/internal/util"
+	"github.com/cyberark/secretless-broker/internal/proxyservice/tcp/pg/protocol"
 )
 
 // ConfigureBackend resolves the backend connection settings and credentials and sets the
 // connectionDetails field.
-func (h *Handler) ConfigureBackend() error {
+func (s *SingleUseConnector) ConfigureBackend(credentials map[string][]byte) error {
 	var err error
-	var options map[string][]byte
 
-	if options, err = h.Resolver.Resolve(h.GetConfig().Credentials); err != nil {
-		return err
-	}
+	keys := reflect.ValueOf(credentials).MapKeys()
```

Comment:
Taken straight from old code...

I understand not using reflection to write our own meta this that and the other, because it's relatively complex and requires a grasp of reflection (which is a bar we're not willing to erect for our developers). There might be a stronger and separate reason, if so please share.

In this case, it is a single line and doesn't seem to require explanation. I'd be inclined to leave it be

Reply:
> Taken straight from old code...Mainly just cause every best practice guide I've come across says to avoid it (eg, https://github.com/codeship/go-best-practices/tree/master/reflection).

_Would you jump off a building if a best practice guide told you to, Jonah?_

Yes.  Yes I would.

But I do actually have some reasons:
1. Since the standard is never to use reflection unless needed, it makes anyone reading the code pause and ask why you did use reflection here.  Is there some reason we're not use using a loop?  Is the code doing more than it seems?
2. Broken windows phenomenon: It gives the stamp approval to this, and then other devs may start using it.  That is, idioms spread inside a code base.
3. I think it makes us look like we don't know